### PR TITLE
csi: add sc privileged to logrotate sidecar container

### PR DESCRIPTION
**Pull request by parth-gr**
deployments on openshift where we need csi container pods to be run on privileged, the logrotate sidecar container was missing the privileged permission letting, side car to not run properly

There was permissions issues with the csi deployments,
Even the cs-logrotate logs mentioned it

```
sh-5.1# cd /var/lib/rook/openshift-storage.cephfs.csi.ceph.com/log/controller-plugin/
sh-5.1# ls
csi-addons.log	csi-cephfsplugin.log
sh-5.1# ls -lart
ls: cannot access 'csi-cephfsplugin.log': Permission denied
ls: cannot access 'csi-addons.log': Permission denied
total 0
-?????????? ? ?    ?     ?            ? csi-cephfsplugin.log
-?????????? ? ?    ?     ?            ? csi-addons.log
drwxr-xr-x. 2 root root 56 Sep 29 17:10 .
drwxr-xr-x. 3 root root 31 Sep 29 17:10 ..
```

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ command:
 image: {{ .CSIPluginImage }}
 imagePullPolicy: IfNotPresent
 name: log-collector
+{{ if and .Privileged .CSILogRotation }}
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml:31]: any specific reason to have `and` check here, can we use `.Privileged` only

Comment (parth-gr) [reply pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml]: yaa e already check for logrotate before appending the container so no need, updated the code

**Comment by Madhu-1** [comment]
@parth-gr can you please update the description in details that what is not working?

**Comment by parth-gr** [comment]
> There was permissions issues with the csi deployments,

Updated

File: pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml (hunk 1 above)
Comment (Madhu-1) [reply pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml]: @parth-gr ^^

Comment (Madhu-1) [reply pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml]: it would be good to add that for kubernetes (vanilla ) as well like we added above

Comment (parth-gr) [reply pkg/operator/ceph/csi/template/csi-logrotate-sidecar.yaml]: Done

**Comment by parth-gr** [comment]
Results:
```
[rider@fedora Downloads]$ oc exec -it csi-cephfsplugin-provisioner-75d98dc76-2dm9b -c log-collector sh 
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
sh-5.1# cd /var/
adm/      db/       ftp/      kerberos/ local/    log/      nis/      preserve/ spool/    yp/       
cache/    empty/    games/    lib/      lock/     mail/     opt/      run/      tmp/      
sh-5.1# cd /var/lib/
alternatives/  cephadm/       dnf/           gssproxy/      misc/          private/       rpcbind/       rpm-state/     selinux/       systemd/       
ceph/          dbus/          games/         logrotate/     nfs/           rook/          rpm/           samba/         smartmontools/ tpm2-tss/      
sh-5.1# cd /var/lib/
alternatives/  cephadm/       dnf/           gssproxy/      misc/          private/       rpcbind/       rpm-state/     selinux/       systemd/       
ceph/          dbus/          games/         logrotate/     nfs/           rook/          rpm/           samba/         smartmontools/ tpm2-tss/      
sh-5.1# cd /var/lib/rook/openshift-storage.cephfs.csi.ceph.com/log
log/              logrotate-config/ 
sh-5.1# cd /var/lib/rook/openshift-storage.cephfs.csi.ceph.com/log/controller-plugin/    
sh-5.1# ls -lh
total 560K
-rw-r--r--. 1 root root 524K Sep 30 14:57 csi-addons.log
-rw-r--r--. 1 root root  30K Sep 30 14:56 csi-cephfsplugin.log
sh-5.1# 

```

**Comment by parth-gr** [comment]
@Madhu-1 please add your reviews

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**